### ISSUE TITLE
Converter issues with Cataloging and Persons

### ISSUE THREAD

**Issue by meganforbes** [collectionspace/cspace-converter#272]
Uploaded all CSV files to this Google folder along with a sheet that notes the error messages I received. Also included copies of CSV files that were successfully uploaded.

https://drive.google.com/drive/folders/1xRIWNT1X_MrtQgV0O1rZKeL_SWaL81it?usp=sharing

**Comment by kspurgin** [comment]
I will try to improve the error handling for the "Import error: invalid byte sequence in UTF-8" error. 

I don't think we want to do anything automatically to "fix" this since it'd almost certainly cause data garbling or loss.

**Comment by kspurgin** [comment]
@meganforbes - can you upload your original "Cataloging CSV First 10" that got the "Import error: invalid byte sequence in UTF-8" error?

I am seeing the "Import error: ERROR: missing headers: objectnumber" but cannot test for the other issue with this file. Thanks!

**Comment by kspurgin** [comment]
With the copy of Cataloging CSV First 10 saved as UTF-8 in Excel, examination of the first row's first cell shows: 

```
2.6.3 :004 > csv[0][0].bytes.map(&:chr)
 => ["\xEF", "\xBB", "\xBF", "o", "b", "j", "e", "c", "t", "N", "u", "m", "b", "e", "r"]
```
`"\xEF", "\xBB", "\xBF"` is the notorious Excel BOM causing trouble as usual. I thought I saw a PR a while back that added handling for Excel BOM, so I will poke at why this is causing a problem now

**Comment by kspurgin** [comment]
@mark-cooper - I'm baffled why the missing header error shows up. 

In IRB this does not throw an error: 

```
SmarterCSV.process(File.open('Cataloging CSV First 10.csv', 'r:bom|utf-8'), { required_headers: %i[objectnumber] })
```

This does FAIL with the: 

```
SmarterCSV.process(File.open('Cataloging CSV First 10.csv', 'r:bom|utf-8'), { auto_row_sep_chars: 10_000, chunk_size: 100, convert_values_to_numeric: false, required_headers: %i[objectnumber], row_sep: :auto })
```

Remove the `row_sep` option and it does NOT fail: 

```
SmarterCSV.process(File.open('Cataloging CSV First 10.csv', 'r:bom|utf-8'), { auto_row_sep_chars: 10_000, chunk_size: 100, convert_values_to_numeric: false, required_headers: %i[objectnumber] })
```

[The SmarterCSV gem docs](https://github.com/tilo/smarter_csv/tree/1.2-stable) says re: `row-sep`: "This can also be set to :auto, but will process the whole cvs file first  (slow!)"

I am guessing we needed the `:auto` option here to support seamless processing of CSVs created on different OSs? 

It seems to me like it's the `File.open` that should be handling/stripping the BOM, and that should be happening before SmarterCSV grabs the file to do anything with it (?)

Side wondering: if `:auto` causes the whole CSV to be processed first, what do we gain by also using chunking?

**Comment by kspurgin** [comment]
I have documented what @meganforbes reported here as the 3 separate issues linked above. I have copied the relevant info from comments to the appropriate new issues. 

Because Megan was able to complete what she was trying to do with these files, I will mark this done. 

The solution that worked was using LibreOffice Calc (free/open source spreadsheet application) to save the files. LibreOffice Calc does not mess with your data and character encoding in problematic ways like Excel does.
